src/cascadia/TerminalControl/TermControl.cpp

```diff
@@ -1274,36 +1274,56 @@ namespace winrt::Microsoft::Terminal::TerminalControl::implementation
         // ComCtl scrollbars, but it's certainly close enough.
         const auto scrollbarSize = GetSystemMetricsForDpi(SM_CXVSCROLL, dpi);
 
-        float width = gsl::narrow<float>(cols * fFontWidth);
+        double width = cols * fFontWidth;
+        double height = rows * fFontHeight;
 
         // Reserve additional space if scrollbar is intended to be visible
         if (settings.ScrollState() == ScrollbarState::Visible)
         {
             width += scrollbarSize;
         }
-
-        const float height = gsl::narrow<float>(rows * fFontHeight);
-
-        return { width, height };
+        
+        // Reserve space for padding
+        const auto paddingArr = _ParsePadding(settings.Padding());
+        switch (paddingArr.size())
+        {
+            case 1:
+                width += paddingArr[0] * 2;
+                height += paddingArr[0] * 2;
+                break;
+            case 2:
+                width += paddingArr[0] * 2;
+                height += paddingArr[1] * 2;
+                break;
+            // No case for 3 padding values, since it's not a norm to provide just Left, Top & Right padding values leaving out Bottom
+            case 4:
+                width += paddingArr[0] + paddingArr[2];
+                height += paddingArr[1] + paddingArr[3];
+                break;
+            default:
+                break;
+        }
+        
+        return { static_cast<float>(width), static_cast<float>(height) };

```

Comment:
nit: gsl::narrow is the preferred cast here.

Reply:
Is `gsl::narrow` preferred? From my understanding and testing, `gsl::narrow` will throw if I set padding to something like 0.3 due to some floating point precision issue.

Reply:
However, I should've use `gsl::narrow_cast` right?

Reply:
gsl::narrow_cast is better than static_cast in my opinion. (I'm not sure that is a requirement or not though by the team)  With that said it looks like the original author went to a lot of trouble to use float math here.   I am not sure if that was on purpose or not.  @zadjii-msft might have an opinion on this.

Reply:
I'm not sure I actually have an opinion here - if `gsl::narrow` will throw for something like .3 but `narrow_cast` won't, then I'd probably use narrow_cast. Usually @adiviness is more of an expert for these sorts of things

Reply:
I think narrow_cast if fine here too.    I haven't pulled this branch and I didn't understand why .3 would cause the value to change.  It could be that my mental compiler is broken though. 😄  I was curious with "some floating point precision issue".   If this is a valid case then I would go with narrow_cast.

Reply:
This is what happened if I set padding to "0, 0.3, 0, 0". std::stod actually put `0.29999999999999999` into the array. ![](https://i.imgur.com/4DOMdFI.png)  
And when `height` was cast to `float`, it became 
![](https://i.imgur.com/hYhD150.png)
Therefore, the two values are not equal and `gsl::narrow` threw.

Reply:
@Summon528 Thanks for the great explanation. (you went beyond what you needed to do - you didn't need to  - this was my responsibility to debug it. 😄  Much appreciated.)  This makes perfect sense.   gsl::narrow for floating point values doesn't make sense because of floating point issues.  So gsl::narrow_cast is the way to go.